src/shims/x86/mod.rs

```diff
@@ -664,6 +725,33 @@ fn convert_float_to_int<'tcx>(
     Ok(())
 }
 
+/// Calculates absolute value of integers in `op` and stores the result in `dest`.
+///
+/// In case of overflow (when the operand is the minimum value), the operation
+/// will wrap around.
+fn int_abs<'tcx>(
+    this: &mut crate::MiriInterpCx<'_, 'tcx>,
+    op: &OpTy<'tcx, Provenance>,
+    dest: &MPlaceTy<'tcx, Provenance>,
+) -> InterpResult<'tcx, ()> {
+    let (op, op_len) = this.operand_to_simd(op)?;
+    let (dest, dest_len) = this.mplace_to_simd(dest)?;
+
+    assert_eq!(op_len, dest_len);
+
+    for i in 0..dest_len {
+        let op = this.read_scalar(&this.project_index(&op, i)?)?;
+        let dest = this.project_index(&dest, i)?;
+
+        // Converting to a host "i128" works since the input is always signed.
+        let res = op.to_int(dest.layout.size)?.unsigned_abs();
```

Comment:
This may be worth an assertion.

Alternatively, a moire clear implementation might avoid using host integers entirely and just use `binop_with_overflow`.

Reply:
I'm not sure what to check with an assertion. How would you implement this with `binop_with_overflow`?

Reply:
The assertion could check that the element type is signed, no?

>  How would you implement this with binop_with_overflow?

Sorry, I meant `wrapping_binary_op`.

Basically run the equivalent of `if op < 0 { -op } else { op }`. It's one `BinOp::lt` and one... okay you need `wrapping_unary_op` as well, `UnOp::Neg`.